// Code generated by mockery v2.30.16. DO NOT EDIT.

package mocks

import (
	fmt "fmt"

	mock "github.com/stretchr/testify/mock"
)

// OrderNotFoundError is an autogenerated mock type for the OrderNotFoundError type
type OrderNotFoundError struct {
	mock.Mock
}

type OrderNotFoundError_Expecter struct {
	mock *mock.Mock
}

func (_m *OrderNotFoundError) EXPECT() *OrderNotFoundError_Expecter {
	return &OrderNotFoundError_Expecter{mock: &_m.Mock}
}

// Cause provides a mock function with given fields:
func (_m *OrderNotFoundError) Cause() error {
	ret := _m.Called()

	var r0 error
	if rf, ok := ret.Get(0).(func() error); ok {
		r0 = rf()
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// OrderNotFoundError_Cause_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Cause'
type OrderNotFoundError_Cause_Call struct {
	*mock.Call
}

// Cause is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) Cause() *OrderNotFoundError_Cause_Call {
	return &OrderNotFoundError_Cause_Call{Call: _e.mock.On("Cause")}
}

func (_c *OrderNotFoundError_Cause_Call) Run(run func()) *OrderNotFoundError_Cause_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_Cause_Call) Return(_a0 error) *OrderNotFoundError_Cause_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_Cause_Call) RunAndReturn(run func() error) *OrderNotFoundError_Cause_Call {
	_c.Call.Return(run)
	return _c
}

// Error provides a mock function with given fields:
func (_m *OrderNotFoundError) Error() string {
	ret := _m.Called()

	var r0 string
	if rf, ok := ret.Get(0).(func() string); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(string)
	}

	return r0
}

// OrderNotFoundError_Error_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Error'
type OrderNotFoundError_Error_Call struct {
	*mock.Call
}

// Error is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) Error() *OrderNotFoundError_Error_Call {
	return &OrderNotFoundError_Error_Call{Call: _e.mock.On("Error")}
}

func (_c *OrderNotFoundError_Error_Call) Run(run func()) *OrderNotFoundError_Error_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_Error_Call) Return(_a0 string) *OrderNotFoundError_Error_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_Error_Call) RunAndReturn(run func() string) *OrderNotFoundError_Error_Call {
	_c.Call.Return(run)
	return _c
}

// Format provides a mock function with given fields: f, verb
func (_m *OrderNotFoundError) Format(f fmt.State, verb rune) {
	_m.Called(f, verb)
}

// OrderNotFoundError_Format_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Format'
type OrderNotFoundError_Format_Call struct {
	*mock.Call
}

// Format is a helper method to define mock.On call
//   - f fmt.State
//   - verb rune
func (_e *OrderNotFoundError_Expecter) Format(f interface{}, verb interface{}) *OrderNotFoundError_Format_Call {
	return &OrderNotFoundError_Format_Call{Call: _e.mock.On("Format", f, verb)}
}

func (_c *OrderNotFoundError_Format_Call) Run(run func(f fmt.State, verb rune)) *OrderNotFoundError_Format_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run(args[0].(fmt.State), args[1].(rune))
	})
	return _c
}

func (_c *OrderNotFoundError_Format_Call) Return() *OrderNotFoundError_Format_Call {
	_c.Call.Return()
	return _c
}

func (_c *OrderNotFoundError_Format_Call) RunAndReturn(run func(fmt.State, rune)) *OrderNotFoundError_Format_Call {
	_c.Call.Return(run)
	return _c
}

// IsCustomError provides a mock function with given fields:
func (_m *OrderNotFoundError) IsCustomError() bool {
	ret := _m.Called()

	var r0 bool
	if rf, ok := ret.Get(0).(func() bool); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(bool)
	}

	return r0
}

// OrderNotFoundError_IsCustomError_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'IsCustomError'
type OrderNotFoundError_IsCustomError_Call struct {
	*mock.Call
}

// IsCustomError is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) IsCustomError() *OrderNotFoundError_IsCustomError_Call {
	return &OrderNotFoundError_IsCustomError_Call{Call: _e.mock.On("IsCustomError")}
}

func (_c *OrderNotFoundError_IsCustomError_Call) Run(run func()) *OrderNotFoundError_IsCustomError_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_IsCustomError_Call) Return(_a0 bool) *OrderNotFoundError_IsCustomError_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_IsCustomError_Call) RunAndReturn(run func() bool) *OrderNotFoundError_IsCustomError_Call {
	_c.Call.Return(run)
	return _c
}

// IsNotFoundError provides a mock function with given fields:
func (_m *OrderNotFoundError) IsNotFoundError() bool {
	ret := _m.Called()

	var r0 bool
	if rf, ok := ret.Get(0).(func() bool); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(bool)
	}

	return r0
}

// OrderNotFoundError_IsNotFoundError_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'IsNotFoundError'
type OrderNotFoundError_IsNotFoundError_Call struct {
	*mock.Call
}

// IsNotFoundError is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) IsNotFoundError() *OrderNotFoundError_IsNotFoundError_Call {
	return &OrderNotFoundError_IsNotFoundError_Call{Call: _e.mock.On("IsNotFoundError")}
}

func (_c *OrderNotFoundError_IsNotFoundError_Call) Run(run func()) *OrderNotFoundError_IsNotFoundError_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_IsNotFoundError_Call) Return(_a0 bool) *OrderNotFoundError_IsNotFoundError_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_IsNotFoundError_Call) RunAndReturn(run func() bool) *OrderNotFoundError_IsNotFoundError_Call {
	_c.Call.Return(run)
	return _c
}

// IsOrderNotFoundError provides a mock function with given fields:
func (_m *OrderNotFoundError) IsOrderNotFoundError() bool {
	ret := _m.Called()

	var r0 bool
	if rf, ok := ret.Get(0).(func() bool); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(bool)
	}

	return r0
}

// OrderNotFoundError_IsOrderNotFoundError_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'IsOrderNotFoundError'
type OrderNotFoundError_IsOrderNotFoundError_Call struct {
	*mock.Call
}

// IsOrderNotFoundError is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) IsOrderNotFoundError() *OrderNotFoundError_IsOrderNotFoundError_Call {
	return &OrderNotFoundError_IsOrderNotFoundError_Call{Call: _e.mock.On("IsOrderNotFoundError")}
}

func (_c *OrderNotFoundError_IsOrderNotFoundError_Call) Run(run func()) *OrderNotFoundError_IsOrderNotFoundError_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_IsOrderNotFoundError_Call) Return(_a0 bool) *OrderNotFoundError_IsOrderNotFoundError_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_IsOrderNotFoundError_Call) RunAndReturn(run func() bool) *OrderNotFoundError_IsOrderNotFoundError_Call {
	_c.Call.Return(run)
	return _c
}

// Message provides a mock function with given fields:
func (_m *OrderNotFoundError) Message() string {
	ret := _m.Called()

	var r0 string
	if rf, ok := ret.Get(0).(func() string); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(string)
	}

	return r0
}

// OrderNotFoundError_Message_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Message'
type OrderNotFoundError_Message_Call struct {
	*mock.Call
}

// Message is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) Message() *OrderNotFoundError_Message_Call {
	return &OrderNotFoundError_Message_Call{Call: _e.mock.On("Message")}
}

func (_c *OrderNotFoundError_Message_Call) Run(run func()) *OrderNotFoundError_Message_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_Message_Call) Return(_a0 string) *OrderNotFoundError_Message_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_Message_Call) RunAndReturn(run func() string) *OrderNotFoundError_Message_Call {
	_c.Call.Return(run)
	return _c
}

// Status provides a mock function with given fields:
func (_m *OrderNotFoundError) Status() int {
	ret := _m.Called()

	var r0 int
	if rf, ok := ret.Get(0).(func() int); ok {
		r0 = rf()
	} else {
		r0 = ret.Get(0).(int)
	}

	return r0
}

// OrderNotFoundError_Status_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Status'
type OrderNotFoundError_Status_Call struct {
	*mock.Call
}

// Status is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) Status() *OrderNotFoundError_Status_Call {
	return &OrderNotFoundError_Status_Call{Call: _e.mock.On("Status")}
}

func (_c *OrderNotFoundError_Status_Call) Run(run func()) *OrderNotFoundError_Status_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_Status_Call) Return(_a0 int) *OrderNotFoundError_Status_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_Status_Call) RunAndReturn(run func() int) *OrderNotFoundError_Status_Call {
	_c.Call.Return(run)
	return _c
}

// Unwrap provides a mock function with given fields:
func (_m *OrderNotFoundError) Unwrap() error {
	ret := _m.Called()

	var r0 error
	if rf, ok := ret.Get(0).(func() error); ok {
		r0 = rf()
	} else {
		r0 = ret.Error(0)
	}

	return r0
}

// OrderNotFoundError_Unwrap_Call is a *mock.Call that shadows Run/Return methods with type explicit version for method 'Unwrap'
type OrderNotFoundError_Unwrap_Call struct {
	*mock.Call
}

// Unwrap is a helper method to define mock.On call
func (_e *OrderNotFoundError_Expecter) Unwrap() *OrderNotFoundError_Unwrap_Call {
	return &OrderNotFoundError_Unwrap_Call{Call: _e.mock.On("Unwrap")}
}

func (_c *OrderNotFoundError_Unwrap_Call) Run(run func()) *OrderNotFoundError_Unwrap_Call {
	_c.Call.Run(func(args mock.Arguments) {
		run()
	})
	return _c
}

func (_c *OrderNotFoundError_Unwrap_Call) Return(_a0 error) *OrderNotFoundError_Unwrap_Call {
	_c.Call.Return(_a0)
	return _c
}

func (_c *OrderNotFoundError_Unwrap_Call) RunAndReturn(run func() error) *OrderNotFoundError_Unwrap_Call {
	_c.Call.Return(run)
	return _c
}

// NewOrderNotFoundError creates a new instance of OrderNotFoundError. It also registers a testing interface on the mock and a cleanup function to assert the mocks expectations.
// The first argument is typically a *testing.T value.
func NewOrderNotFoundError(t interface {
	mock.TestingT
	Cleanup(func())
}) *OrderNotFoundError {
	mock := &OrderNotFoundError{}
	mock.Mock.Test(t)

	t.Cleanup(func() { mock.AssertExpectations(t) })

	return mock
}
